docs/source/python/integration/python_java.rst

```diff
@@ -0,0 +1,429 @@
+
+.. code-block:: xml
+
+    <project>
+        <modelVersion>4.0.0</modelVersion>
+        
+        <groupId>org.apache.arrow.py2java</groupId>
+        <artifactId>FillTen</artifactId>
+        <version>1</version>
+
+        <properties>
+            <maven.compiler.source>1.7</maven.compiler.source>
+            <maven.compiler.target>1.7</maven.compiler.target>
+        </properties> 
+
+        <dependencies>
+            <dependency>
+            <groupId>org.apache.arrow</groupId>
+            <artifactId>arrow-memory</artifactId>
+            <version>8.0.0-SNAPSHOT</version>
+            <type>pom</type>
+            </dependency>
+            <dependency>
+            <groupId>org.apache.arrow</groupId>
+            <artifactId>arrow-memory-netty</artifactId>
+            <version>8.0.0-SNAPSHOT</version>
+            <type>jar</type>
+            </dependency>
+            <dependency>
+            <groupId>org.apache.arrow</groupId>
+            <artifactId>arrow-vector</artifactId>
+            <version>8.0.0-SNAPSHOT</version>
+            <type>pom</type>
+            </dependency> 
+            <dependency>
+            <groupId>org.apache.arrow</groupId>
+            <artifactId>arrow-c-data</artifactId>
+            <version>8.0.0-SNAPSHOT</version>
+            <type>jar</type>
+            </dependency>
+        </dependencies>
+    </project>
+
+Once the ``FillTen.java`` file with the class is created
+as ``src/main/java/FillTen.java`` we can use ``maven`` to
+compile the project with ``mvn package`` and get it 
+available in the ``target`` directory.
+
+.. code-block:: bash
+
+    $ mvn package
+    [INFO] Scanning for projects...
+    [INFO] 
+    [INFO] ------------------< org.apache.arrow.py2java:FillTen >------------------
+    [INFO] Building FillTen 1
+    [INFO] --------------------------------[ jar ]---------------------------------
+    [INFO] 
+    [INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ FillTen ---
+    [INFO] Changes detected - recompiling the module!
+    [INFO] Compiling 1 source file to /experiments/java2py/target/classes
+    [INFO] 
+    [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ FillTen ---
+    [INFO] Building jar: /experiments/java2py/target/FillTen-1.jar
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------
+
+Now that we have the package built, we can make it available to Python.
+To do so, we need to make sure that not only the package itself is available,
+but that also its dependencies are too.
+
+We can use ``maven`` to collect all dependencies and make them available in a single place
+(the ``dependencies`` directory) so that we can more easily load them from Python:
+
+.. code-block:: bash
+
+    $ mvn org.apache.maven.plugins:maven-dependency-plugin:2.7:copy-dependencies -DoutputDirectory=dependencies
```

Comment:
I'm clueless about Java, but can't we just produce a single jar and give it to jpype?

Reply:
Yes, we can using a maven plugin. But I wanted to make explicit the step where you need to collect and load dependencies by yourself. I'll add a note that the alternative is to use `maven-assembly-plugin` to assemble a single jar with all dependencies inside.